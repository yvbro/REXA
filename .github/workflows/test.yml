name: tests

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  build:
    runs-on: ubuntu-latest
    services:
      postgres:
        image: postgres:12-alpine
        env:
          POSTGRES_USER: rexaAdmin
          POSTGRES_PASSWORD: rexaAdmin
          POSTGRES_DB: rexa
          POSTGRES_HOST: localhost
        ports:
          - 5442:5432
        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5

    steps:
      - uses: actions/checkout@v2
      - name: Set up JDK 11
        uses: actions/setup-java@v1
        with:
          java-version: 11
      - name: Init db
        run: mvn flyway:migrate
      - name: Unit test and integration test
        run: mvn -B clean verify --file pom.xml