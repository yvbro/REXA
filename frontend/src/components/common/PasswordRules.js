import React from 'react';

import { Typography } from '@material-ui/core';

const PasswordRules = () => {
    return (<Typography variant="caption" display="block" gutterBottom>
            The password must be 8 characters long.
            It should contain a capital letter and a number.
        </Typography>
    );
};

export default PasswordRules;