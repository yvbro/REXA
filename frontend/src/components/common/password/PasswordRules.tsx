import React from 'react';

import { Typography } from '@material-ui/core';

const PasswordRules = () => {
    return (
        <Typography
            data-testid="passwordRules"
            variant="caption"
            display="block"
            gutterBottom
        >
            The password must be 8 characters long. It should contain a capital
            letter and a number.
        </Typography>
    );
};

export default PasswordRules;
