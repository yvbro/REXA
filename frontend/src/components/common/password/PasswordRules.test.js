import React from 'react';
import { cleanup, renderWithStore } from '../../../helpers/test/test-utils';
import '@testing-library/jest-dom/extend-expect';

import PasswordRules from './PasswordRules';

describe('The PasswordRules component', () => {
    afterEach(cleanup);

    it('should take a snapshot', () => {
        const { asFragment } = renderWithStore(<PasswordRules />, {});

        expect(asFragment(<PasswordRules />)).toMatchSnapshot();
    });

    it('should display the password rules', () => {
        const { getByTestId } = renderWithStore(<PasswordRules />, {});

        expect(getByTestId('passwordRules')).toHaveTextContent(
            'The password must be 8 characters long. It should contain a capital letter and a number.'
        );
    });
});
