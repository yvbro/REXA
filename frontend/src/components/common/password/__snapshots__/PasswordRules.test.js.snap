// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The PasswordRules component should take a snapshot 1`] = `
<DocumentFragment>
  <span
    class="MuiTypography-root MuiTypography-caption MuiTypography-gutterBottom MuiTypography-displayBlock"
    data-testid="passwordRules"
  >
    The password must be 8 characters long. It should contain a capital letter and a number.
  </span>
</DocumentFragment>
`;
